File: backend-laravel/app/Models/Alert.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUlids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Alert extends Model
{
    use HasFactory, HasUlids;

    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [
        'alert_id',
        'device_id',
        'severity',
        'category',
        'message',
        'timestamp',
        'acknowledged',
    ];

    protected $casts = [
        'timestamp' => 'datetime',
        'acknowledged' => 'boolean',
    ];
}

File: mobile-app/lib/models/alert.dart
class AlertItem {
  const AlertItem({
    required this.alertId,
    required this.deviceId,
    required this.severity,
    required this.category,
    required this.message,
    required this.timestamp,
    required this.acknowledged,
  });

  final String? alertId;
  final String? deviceId;
  final String? severity;
  final String? category;
  final String? message;
  final String? timestamp;
  final bool acknowledged;

  factory AlertItem.fromJson(Map<String, dynamic> json) {
    return AlertItem(
      alertId: json['alert_id'] as String?,
      deviceId: json['device_id'] as String?,
      severity: json['severity'] as String?,
      category: json['category'] as String?,
      message: json['message'] as String?,
      timestamp: json['timestamp'] as String?,
      acknowledged: json['acknowledged'] as bool? ?? false,
    );
  }
}
